sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -194,6 +194,66 @@ abstract class JdbcDialect extends Serializable with Logging{
     case _ => value
   }
 
+  protected final val BINARY_COMPARISON = Set("=", "!=", "<=>", "<", "<=", ">", ">=",
+    "+", "-", "*", "/", "%", "&&", "||", "AND", "OR", "&", "|", "^")
+
+  /**
+   * Converts V2 expression to String representing a SQL expression.
+   * @param expr The V2 expression to be converted.
+   * @return Converted value.
+   */
+  @Since("3.3.0")
+  def compileExpression(expr: Expression): Option[String] = expr match {
+    case l @ LiteralValue(_, _) => Some(l.toString)
```

Comment:
```suggestion
    case l: LiteralValue => Some(l.toString)
```

Reply:
should avoid `l` https://github.com/databricks/scala-style-guide#variable-naming-convention as a variable name.

Reply:
Yes. Thank you for your reminder.
But `l: Literal` used in many places.
https://github.com/apache/spark/blob/1c0793a75b74ac7b55631e61d9e827e93647f1d2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DecimalPrecision.scala#L325

Reply:
Yeah. There was a try to repalce them all but it was rejected as it's too invasive. We should better avoid them anyway.